### ISSUE TITLE
Connection reset by peer 

### ISSUE THREAD

**Issue by thecb4** [fastlane/fastlane#6553]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

##### Complete output when running fastlane, including the stack trace and command used

```
+-----------------------+-----------------------------------------------+
|                        Summary for match 0.8.1                        |
+-----------------------+-----------------------------------------------+
| git_url               | git@gitlab.com:thecb4/XXXXXX.git |
| app_identifier        | io.thecb4.XXXXXX                |
| type                  | appstore                                      |
| verbose               | true                                          |
| git_branch            | master                                        |
| username              | cavelle@XXXXX                            |
| keychain_name         | login.keychain                                |
| readonly              | false                                         |
| team_id               | XXXXXX                                    |
| force                 | false                                         |
| shallow_clone         | false                                         |
| force_for_new_devices | false                                         |
| skip_docs             | false                                         |
+-----------------------+-----------------------------------------------+
```

```
INFO [2016-10-14 16:52:47.88]: Cloning remote git repo...
INFO [2016-10-14 16:52:47.89]: $ GIT_TERMINAL_PROMPT=0 git clone 'git@gitlab.com:thecb4/FastlaneMatchTheCB4.git' '/var/folders/yh/rbjzzgk50b7f5g8pnyrntf8r0000gn/T/d20161014-9928-1ycj0u6'
INFO [2016-10-14 16:52:47.91]: ▸ Cloning into '/var/folders/yh/rbjzzgk50b7f5g8pnyrntf8r0000gn/T/d20161014-9928-1ycj0u6'...
INFO [2016-10-14 16:52:49.50]: ▸ remote: Counting objects: 57, done.
INFO [2016-10-14 16:52:49.69]: ▸ remote: Compressing objects: 100% (53/53), done.
INFO [2016-10-14 16:52:49.84]: ▸ remote: Total 57 (delta 4), reused 0 (delta 0)
INFO [2016-10-14 16:52:49.84]: ▸ Receiving objects: 100% (57/57), 102.94 KiB | 0 bytes/s, done.
INFO [2016-10-14 16:52:49.84]: ▸ Resolving deltas: 100% (4/4), done.
INFO [2016-10-14 16:52:49.85]: ▸ Checking connectivity... done.
INFO [2016-10-14 16:52:49.97]: 🔓  Decrypted '--------.cer'
INFO [2016-10-14 16:52:50.06]: 🔓  Decrypted '--------.cer'
INFO [2016-10-14 16:52:50.15]: 🔓  Decrypted '--------.p12'
INFO [2016-10-14 16:52:50.25]: 🔓  Decrypted '--------.p12'
INFO [2016-10-14 16:52:50.34]: 🔓  Decrypted 'AppStore_io.thecb4.----.mobileprovision'
INFO [2016-10-14 16:52:50.43]: 🔓  Decrypted 'AppStore_io.thecb4.----.mobileprovision'
INFO [2016-10-14 16:52:50.52]: 🔓  Decrypted 'Development_io.thecb4.----.mobileprovision'
INFO [2016-10-14 16:52:50.62]: 🔓  Decrypted 'Development_io.thecb4.----.mobileprovision'
INFO [2016-10-14 16:52:50.62]: 🔓  Successfully decrypted certificates repo
INFO [2016-10-14 16:52:50.69]: Verifying that the certificate and profile are still valid on the Dev Portal...
INFO [2016-10-14 16:53:05.22]: 🔒  Successfully encrypted certificates repo
INFO [2016-10-14 16:53:05.23]: -------------------
INFO [2016-10-14 16:53:05.23]: --- Step: slack ---
INFO [2016-10-14 16:53:05.23]: -------------------
INFO [2016-10-14 16:53:05.68]: Successfully sent Slack notification
WARN [2016-10-14 16:53:05.68]: Variable Dump:
INFO [2016-10-14 16:53:05.68]: {:DEFAULT_PLATFORM=>:ios, :PLATFORM_NAME=>:ios, :LANE_NAME=>"ios release_internal_beta", :GIT_REPO_WAS_CLEAN_ON_START=>true}
ERROR [2016-10-14 16:53:05.68]: Connection reset by peer - SSL_connect
```

### Configuration Files

Please copy the complete content of your `Fastfile` and any other configuration files you use below:

Fastfile:
```ruby
  lane :bootstrap_provisions do |options|

    provision_type = "development"

    if options[:type] != nil
      provision_type = options[:type]
    end

    match(
      git_url: "git@gitlab.com:thecb4/----.git",
      app_identifier: "io.thecb4.----,
      type: provision_type,
      # readonly: true
      verbose: true
    )
  end
```

### Environment

fastlane version (run `fastlane -v`):
fastlane 1.105.2
match 0.8.1

Do you use bundler to execute fastlane (i.e. `bundle exec fastlane`)? yes

Do you use a Ruby environment manager (e.g. `chruby`, `rbenv`, `rvm`)? no

What Xcode version do you use (e.g. 7.3, 8.0) 8.0

**Comment by johnryan** [comment]
The same just started happening for me, i've tried several things based on previous answers (reinstalling openSSL for instance) but no luck yet.

**Comment by thecb4** [comment]
Ok.  I feel sane now.  It was fine yesterday and now it doesn't like me.

**Comment by klundberg** [comment]
This started happening earlier today for us with cert as well, so it may not just be match with the problem

**Comment by lukethoene** [comment]
Yep, it's happening during sigh for us 

{:DEFAULT_PLATFORM=>:ios, :PLATFORM_NAME=>:ios, :LANE_NAME=>"ios publisher"}
[14:13:32]: Connection reset by peer - SSL_connect

**Comment by pcsaito** [comment]
Same here, after cleaning all profiles it worked only once

INFO [2016-10-14 18:13:56.53]: Verifying that the certificate and profile are still valid on the Dev Portal...
INFO [2016-10-14 18:14:15.98]: 🔒  Successfully encrypted certificates repo
WARN [2016-10-14 18:14:15.98]: Variable Dump:
INFO [2016-10-14 18:14:15.98]: {:DEFAULT_PLATFORM=>:ios, :PLATFORM_NAME=>:ios, :LANE_NAME=>"ios certificates"}
ERROR [2016-10-14 18:14:15.99]: Connection reset by peer - SSL_connect

**Comment by cava23** [comment]
+1

**Comment by aronse** [comment]
+1

**Comment by raphaelbaldi** [comment]
+1

**Comment by gaylejj** [comment]
Having this issue with "fastlane init" as well when it tries to use produce to create an app

**Comment by craigho** [comment]
+1 -- apple issue?

**Comment by mofojed** [comment]
+1

**Comment by GlaucoNeves** [comment]
+1

**Comment by dpassage** [comment]
So we've discovered that this failure occurs if you use the system Ruby, i.e. /usr/bin/ruby, on OS X 10.11.6. Using Ruby 2.3.0 installed via RVM, this works fine.

**Comment by davidgovea** [comment]
Thanks @dpassage! Installed ruby 2.3.0 via RVM, reinstalled fastlane gem, and we're up and running again.

**Comment by rvi** [comment]
Same here, I _just_ update my ruby version via `rbenv`:

```
brew update
brew install rbenv
rbenv install 2.2.4
rbenv global 2.2.4
```

Then you may need to modify your `.bash_profile` or `.zshrc` if you use zsh.
Add these 2 lines:
```
export PATH="$HOME/.rbenv/bin:$PATH"
eval "$(rbenv init -)"
```

It should work after a reboot :)

**Comment by dpassage** [comment]
I haven't dug into this further, but I'm guessing that Apple changed what SSL ciphers are acceptable on the dev portal to something not used by default by the system ruby.

**Comment by gaylejj** [comment]
@dpassage Thanks a lot! Seems to be working great now.

**Comment by zineer** [comment]
Thanks for the fix!
```
brew install ruby
gem install fastlane
```
got me back in business

**Comment by dpassage** [comment]
It appears that https://developer.apple.com is no longer accepting TLS versions below 1.2:

```
$ curl --tlsv1.1 https://developer.apple.com
curl: (35) Server aborted the SSL handshake
$ curl --tlsv1.0 https://developer.apple.com
curl: (35) Server aborted the SSL handshake
```

Unfortunately, it appears that the system ruby on El Capitan doesn't support 1.2. :(

@KrauseFx looks like `spaceship` and thus `match` now have a dependency on newer versions of ruby :(. Or some major surgery to make `faraday` use a different SSL library.

**Comment by JotaMelo** [comment]
It started happening this afternoon for me, @zineer saved the day.

**Comment by pedro380085** [comment]
+1

**Comment by thecb4** [comment]
Configuring for a different Ruby/Bundle took a lot of effort for some reason.  Would be good to get some clean instructions.

**Comment by Nimothy** [comment]
Tried updating ruby: 
`ruby 2.3.1p112 (2016-04-26 revision 54768) [x86_64-darwin16]`

Still getting the same error, only happened after I ran:
`bundle update`

**Comment by hjanuschka** [comment]
have you guys tried:

```
rvm reinstall ruby-2.2.5 --with-openssl-dir=/usr/local
```

**Comment by nthegedus** [comment]
No one of this methods above worked with me =\

I still getting the "Connection reset by peer - SSL_connect" message

**Comment by JonasWestAlro** [comment]
+1

**Comment by rompic** [comment]
+1

**Comment by lifuzu** [comment]
@zineer way works for me, thanks!

**Comment by pascalc** [comment]
After using `rbenv` to update ruby to version 2.3 and reinstalling `fastlane`, I had to do `bundle exec fastlane` for it to work again.

**Comment by JimHowell20** [comment]
+1

**Comment by JimHowell20** [comment]
+1 none of the fixes have worked so far for me

**Comment by thecb4** [comment]
Seems to work for me

```shell
$ rbenv install 2.3.1
$ rbenv global 2.3.1
$ gem install bundler -v 1.13.4
$ bundle install --path ./.gems --verbose
$ bundle exec fastlane --verbose your_lane
```

**Comment by fuchanghuan** [comment]
@zineer works for me,Thanks.

**Comment by getreup** [comment]
+1 

none of the steps above have helped.

**Comment by jk2K** [comment]
@zineer, There is another error now
```
Looking for related GitHub issues on fastlane/fastlane...

➡️  Version getters
   https://github.com/fastlane/fastlane/pull/504 [closed] 4 💬
   17 Aug 2015

/usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/net/http/response.rb:120:in `error!': [!] 504 "Connection failed" (Faraday::ConnectionFailed)
	from /usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/net/http/response.rb:129:in `value'
	from /usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/net/http.rb:920:in `connect'
	from /usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/net/http.rb:863:in `do_start'
	from /usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/net/http.rb:852:in `start'
	from /usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/net/http.rb:1398:in `request'
	from /usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/net/http.rb:1156:in `get'
	from /usr/local/lib/ruby/gems/2.3.0/gems/faraday-0.9.2/lib/faraday/adapter/net_http.rb:80:in `perform_request'
	from /usr/local/lib/ruby/gems/2.3.0/gems/faraday-0.9.2/lib/faraday/adapter/net_http.rb:40:in `block in call'
	from /usr/local/lib/ruby/gems/2.3.0/gems/faraday-0.9.2/lib/faraday/adapter/net_http.rb:87:in `with_net_http_connection'
	from /usr/local/lib/ruby/gems/2.3.0/gems/faraday-0.9.2/lib/faraday/adapter/net_http.rb:32:in `call'
	from /usr/local/lib/ruby/gems/2.3.0/gems/faraday-cookie_jar-0.0.6/lib/faraday/cookie_jar.rb:22:in `call'
	from /usr/local/lib/ruby/gems/2.3.0/gems/faraday_middleware-0.10.0/lib/faraday_middleware/response_middleware.rb:30:in `call'
	from /usr/local/lib/ruby/gems/2.3.0/gems/faraday_middleware-0.10.0/lib/faraday_middleware/response_middleware.rb:30:in `call'
	from /usr/local/lib/ruby/gems/2.3.0/gems/faraday_middleware-0.10.0/lib/faraday_middleware/response_middleware.rb:30:in `call'
	from /usr/local/lib/ruby/gems/2.3.0/gems/faraday-0.9.2/lib/faraday/rack_builder.rb:139:in `build_respon
```

**Comment by medslt** [comment]
+1

**Comment by tefinger** [comment]
+1

**Comment by iTaa** [comment]
+1

**Comment by chstraube** [comment]
+1

**Comment by medslt** [comment]
The same problem here when building with enterprise account (worked fine with App Store account)

**Comment by rompic** [comment]
@zineer thx. also had to gem update --system before in order to fix https://github.com/rubygems/rubygems/issues/1560 'no implicit conversion of nil into String' when trying to install fastlane.

**Comment by joamafer** [comment]
I had the same problem and I fixed by doing the following:
1) Follow the instructions here (https://developer.xamarin.com/guides/testcloud/calabash/configuring/osx/updating-ruby-using-rbenv/) to install _rbenv_ and _ruby-build_.
2) Run `$ rbenv install 2.3.1` and `rbenv global 2.3.1`.
3) Reinstall fastlane `gem install fastlane`.
4) Restart the build server.

**Comment by medslt** [comment]
I fixed the problem by updating ruby:

brew install rbenv ruby-build

# Add rbenv to bash so that it loads every time you open a terminal
echo 'if which rbenv > /dev/null; then eval "$(rbenv init -)"; fi' >> ~/.bash_profile
source ~/.bash_profile

# Install Ruby
rbenv install 2.3.1
rbenv global 2.3.1
ruby -v

**Comment by danielbas** [comment]
@thecb4 Thanks! It works!

**Comment by dosgroup** [comment]
Doesn't work for me. It continues to use Ruby version 2.0.0 (system):

/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/net/http.rb:921:in `connect': [!] Connection reset by peer - SSL_connect (Faraday::ConnectionFailed)

**Comment by thecb4** [comment]
@dosgroup install local gems and use bundler with local installs.  That was the only way it worked for me.

**Comment by joamafer** [comment]
@dosgroup have you tried to install Ruby 2.3.1 following the instructions above? Don't forget to call this command too: `rbenv global 2.3.1` and check the version with `ruby -v`

**Comment by dosgroup** [comment]
@thecb4 @joamafer I solved by using rvm. Thanks anyway!

**Comment by micnguyen** [comment]
rbenv trick works!

**Comment by hjanuschka** [comment]
for rvm users.

replace with your ruby version

```
brew upgrade openssl
brew link --force openssl
rvm osx-ssl-certs update all
rvm reinstall ruby-2.1.5 --with-openssl-dir=/usr/local
```

**Comment by sharkusmarkus** [comment]
Thanks to all for the steps on upgrading Ruby. Two of my boxes did not exhibit the issue, one did, and of course it was the build server! Ran a quick test post ruby and fastlane update and it seems to be fixed.

**Comment by flaushi** [comment]
I'm lost here.
Tried medslt's rbenv commands, no luck.
brew installed ruby, no luck. now all my gems are gone and i can't install new ones, only getting 
$ sudo gem install fastlane
ERROR:  While executing gem ... (TypeError)
    no implicit conversion of nil into String

**Comment by rafaelnobrepd** [comment]
Not working here either. Upgrade to 2.3.1 ruby from brew and tried using rbenv instead, but still get
`/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/net/http.rb:921:in `connect': [!] Connection reset by peer - SSL_connect (Faraday::ConnectionFailed)`

**Comment by micnguyen** [comment]
Just a heads up - for those that are installing rbenv, make sure you reinstall gem install fastlane after switching to rbenv.

**Comment by vkaltchev** [comment]
Same here. Tried brew, rbenv, rvm.
Now, when trying to gem install fastlane:
ERROR: While executing gem ... (TypeError)
no implicit conversion of nil into String

**Comment by rafaelnobrepd** [comment]
Ok, that was it. `gem install fastlane` right after upgrading ruby did it.

**Comment by hjanuschka** [comment]
also check that openssl is up to date, and has recent root-ca's

```
cd /usr/local/etc/openssl/
curl -O http://curl.haxx.se/ca/cacert.pem
mv cacert.pem cert.pem
```

**Comment by flaushi** [comment]
Sorry, that does not work for me:
$ gem install fastlane
ERROR:  While executing gem ... (Errno::EACCES)
    Permission denied @ rb_sysopen - /usr/local/lib/ruby/gems/2.3.0/gems/commander-4.4.0/.gitignore

sudo gem install fastlane
Password:
ERROR:  While executing gem ... (TypeError)
    no implicit conversion of nil into String

**Comment by vkaltchev** [comment]
@flaushi had the same error when I was also adding the directory - gem install -n /usr/local/bin fastlane, when I ran just "gem install fastlane" it worked fine... Are you sure you're not adding something too?

**Comment by flaushi** [comment]
Yes I am. But I am not experienced with ruby.
When I tried that rbenv-thing, I had to edit my .bash_profile, where I saw these entires, maybe they influence my problems:
PATH=/usr/local/sbin:~/code/android-sdk-macosx/platform-tools:~/.gems/bin:$PATH
GEM_HOME=~/.gems

**Comment by rafaelnobrepd** [comment]
@flaushi try `sudo chown -R $(whoami) /usr/local/lib/ruby/gems`

**Comment by flaushi** [comment]
Hmm, there seems to be a larger system change involved.
My old (ruby 2.0.0) gems were located at /Library/Ruby/Gems/2.0.0/gems/
After brew install ruby, it tries to install them to  /usr/local/lib/ruby/gems/2.3.0/gems/

**Comment by hjanuschka** [comment]
@flaushi system ruby uses `/Library/`  brew ruby uses `/usr/local/` thats normal

**Comment by jk2K** [comment]
@zineer, the error is automatically gone, now it work for me, thanks

`/usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/net/http/response.rb:120:in `error!': [!] 504 "Connection failed" (Faraday::ConnectionFailed)`

**Comment by tamastimar** [comment]
Will it be addressed in a @fastlane release or it should be solved by updating the env?

**Comment by flaushi** [comment]
Ok, more information,
$ gem install fastlane --debug --backtrace --verbose
NOTE:  Debugging mode prints all exceptions even when rescued
Exception `LoadError' at /usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/resolv.rb:171 - LoadError
HEAD https://api.rubygems.org/api/v1/dependencies
200 OK
GET https://api.rubygems.org/api/v1/dependencies?gems=fastlane
200 OK
Exception `Resolv::DNS::Config::NXDomain' at /usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/resolv.rb:549 - _rubygems._tcp.api.rubygems.org
Exception `Resolv::DNS::Config::NXDomain' at /usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/resolv.rb:549 - _rubygems._tcp.api.rubygems.org.fritz.box
Exception `Resolv::ResolvError' at /usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/resolv.rb:494 - DNS result has no information for _rubygems._tcp.api.rubygems.org
Getting SRV record failed: DNS result has no information for _rubygems._tcp.api.rubygems.org
GET https://api.rubygems.org/api/v1/dependencies?gems=activesupport,addressable,bundler,cert,credentials_manager,deliver,fastlane_core,frameit,gym,krausefx-shenzhen,match,multipart-post,pem,pilot,plist,produce,scan,screengrab,sigh,slack-notifier,snapshot,spaceship,supply,terminal-notifier,terminal-table,word_wrap,xcode-install,xcodeproj,xcpretty
200 OK
GET https://api.rubygems.org/api/v1/dependencies?gems=i18n,json,minitest,thread_safe,tzinfo
200 OK
GET https://api.rubygems.org/api/v1/dependencies?gems=claide,colored,unicode-display_width
200 OK
GET https://api.rubygems.org/api/v1/dependencies?gems=babosa,commander,excon,faraday,faraday-cookie_jar,faraday_middleware,fastimage,gh_inspector,google-api-client,highline,multi_json,multi_xml,rubyzip,security
200 OK
GET https://api.rubygems.org/api/v1/dependencies?gems=googleauth,http-cookie,httpclient,hurley,memoist,mime-types,representable,retriable,rouge,xcpretty-travis-formatter
200 OK
GET https://api.rubygems.org/api/v1/dependencies?gems=dotenv,mime-types-data,net-sftp
200 OK
GET https://api.rubygems.org/api/v1/dependencies?gems=domain_name,jwt,logging,os,signet
200 OK
GET https://api.rubygems.org/api/v1/dependencies?gems=unf
200 OK
GET https://api.rubygems.org/api/v1/dependencies?gems=unf_ext
200 OK
GET https://api.rubygems.org/api/v1/dependencies?gems=mini_magick,uber
200 OK
GET https://api.rubygems.org/api/v1/dependencies?gems=little-plugger,net-ssh
200 OK
Exception `Errno::EEXIST' at /usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/fileutils.rb:253 - File exists @ dir_s_mkdir - /usr/local/lib/ruby/gems/2.3.0
Exception `Errno::ENOENT' at /usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/fileutils.rb:1451 - No such file or directory @ unlink_internal - /usr/local/lib/ruby/gems/2.3.0/specifications/commander-4.4.0.gemspec
Exception `Errno::EEXIST' at /usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/fileutils.rb:253 - File exists @ dir_s_mkdir - /usr/local/lib/ruby/gems/2.3.0/gems/commander-4.4.0
Exception `Errno::ENOENT' at /usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/fileutils.rb:1331 - No such file or directory @ rb_file_s_lstat - /usr/local/lib/ruby/gems/2.3.0/gems/commander-4.4.0/.gitignore
Exception `Errno::ENOENT' at /usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/fileutils.rb:1331 - No such file or directory @ rb_file_s_lstat - /usr/local/lib/ruby/gems/2.3.0/gems/commander-4.4.0/.gitignore
Exception `Errno::ENOENT' at /usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/fileutils.rb:1451 - No such file or directory @ unlink_internal - /usr/local/lib/ruby/gems/2.3.0/gems/commander-4.4.0/.gitignore
Exception `Errno::EEXIST' at /usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/fileutils.rb:253 - File exists @ dir_s_mkdir - /usr/local/lib/ruby/gems/2.3.0/gems/commander-4.4.0
/usr/local/lib/ruby/gems/2.3.0/gems/commander-4.4.0/.gitignore
Exception `NoMethodError' at /usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/rubygems/package/tar_reader.rb:72 - undefined method `seek' for #<Zlib::GzipReader:0x007fd862acadb8>
[---snip---]
Exception `Errno::ENOENT' at /usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/fileutils.rb:1331 - No such file or directory @ rb_file_s_lstat - /usr/local/lib/ruby/gems/2.3.0/gems/commander-4.4.0/spec/ui_spec.rb
Exception `Errno::ENOENT' at /usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/fileutils.rb:1451 - No such file or directory @ unlink_internal - /usr/local/lib/ruby/gems/2.3.0/gems/commander-4.4.0/spec/ui_spec.rb
Exception `Errno::EEXIST' at /usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/fileutils.rb:253 - File exists @ dir_s_mkdir - /usr/local/lib/ruby/gems/2.3.0/gems/commander-4.4.0/spec
/usr/local/lib/ruby/gems/2.3.0/gems/commander-4.4.0/spec/ui_spec.rb
Exception `NoMethodError' at /usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/rubygems/package/tar_reader.rb:72 - undefined method `seek' for #<Zlib::GzipReader:0x007fd862acadb8>
Exception `TypeError' at /usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/rubygems/installer.rb:227 - no implicit conversion of nil into String
ERROR:  While executing gem ... (TypeError)
    no implicit conversion of nil into String
	/usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/rubygems/installer.rb:227:in `check_executable_overwrite'
	/usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/rubygems/installer.rb:468:in `block in generate_bin'
	/usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/rubygems/installer.rb:455:in `each'
	/usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/rubygems/installer.rb:455:in `generate_bin'
	/usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/rubygems/installer.rb:295:in `install'
	/usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/rubygems/request_set.rb:167:in `block in install'
	/usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/rubygems/request_set.rb:151:in `each'
	/usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/rubygems/request_set.rb:151:in `install'
	/usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/rubygems/commands/install_command.rb:250:in `install_gem'
	/usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/rubygems/commands/install_command.rb:300:in `block in install_gems'
	/usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/rubygems/commands/install_command.rb:296:in `each'
	/usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/rubygems/commands/install_command.rb:296:in `install_gems'
	/usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/rubygems/commands/install_command.rb:203:in `execute'
	/usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/rubygems/command.rb:308:in `invoke_with_build_args'
	/usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/rubygems/command_manager.rb:169:in `process_args'
	/usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/rubygems/command_manager.rb:139:in `run'
	/usr/local/Cellar/ruby/2.3.1_2/lib/ruby/2.3.0/rubygems/gem_runner.rb:55:in `run'
	/usr/local/bin/gem:21:in `<main>'

**Comment by kd8ssq** [comment]
Thanks @joamafer for the steps.  They worked great!

**Comment by flaushi** [comment]
ok, i got it.
brew uninstall ruby # not using homebrew's ruby

rm -rf ~/gem
rm -rf ~/gems
rm -rf ~/rbenv
brew install rbenv ruby-built
# Add rbenv to bash so that it loads every time you open a terminal
echo 'if which rbenv > /dev/null; then eval "$(rbenv init -)"; fi' >> ~/.bash_profile
source ~/.bash_profile

# Install Ruby
rbenv install 2.3.1
rbenv global 2.3.1

# close teminal and re-open it
ruby -v # should output 2.3.1 now
sudo chown -R $(whoami): ~/.rbenv # important!!!
gem update --system
gem install fastlane # this will install a lot of gems, since we are not using the old 2.0.0 gems anymore.

**Comment by KrauseFx** [comment]
Thanks @zineer for providing a solution, could everyone try the following and let us know if that fixes the issue for you? 

```
brew install ruby
gem install fastlane
```

If so, we'll update fastlane, to properly catch the error, and tell the user to run the above shown command.

**Comment by wilrnh** [comment]
I just ran the above with much success 👍

**Comment by ericertl** [comment]
@flaushi instructions worked great.
The only issue I had was while running `rbenv install 2.3.1`
Looks like when I updated xcode, I didn't install xcode command line tools, so running `xcode-select --install` fixed it

**Comment by PavelChernyshov** [comment]
Using ruby 2.0.0 and spaceship 0.36.0.

Wrote a small script. In my case only the download profile phase fails. 
```ruby
...
profiles = Spaceship.provisioning_profile.all 
profiles.each { |profile| puts profile.uuid }
puts 'trying to download'
profiles.first.download
```
Output: 
```c18473d1-8151-4ab8-b203-c9ba07c7e78f
acc4058d-4a5a-4b0b-984e-fb37d0bd92ac
d62d83e9-e15a-4471-bd77-01ce9955888d
6d848927-676d-45cf-b2ca-0ba94f25e6c2
a74c57c4-bb40-4205-bc8d-a690e61553b4
82cac19e-dd18-4fc3-89d5-585c6023854f
2b4ca520-9ff9-4ea8-a46d-a5a2aef8afb9
a36d36aa-0477-4f7c-8dd3-f0ed9d264a75
d8518fd3-79fa-40ca-ac10-65d107e57415
b452889e-2af7-47a4-9641-2b9795aff7d0
5c8582ad-932e-4eb5-b416-58599bc47620
71aad379-f265-47ca-92ed-65768a79a94a
b283ce24-da61-44de-9bdf-15280efa6f3b
3df581dc-9e6e-4128-883c-36065e356b4f
808f2859-bb5a-49da-8cc3-1184a030af82
1c8576e2-4ba7-4ca2-98df-86d6c2fc5d4c
6d8fb8aa-e84c-4b08-b188-9181e8609763
a954bd0f-a4b5-498f-a80d-c4c6d7b9d2a0
900312d0-1aca-412e-834a-4326c0689c88
0c4405e1-d803-4891-97fe-f5948f5a3444
ca29eee0-be14-45b5-a098-9ba037a3b92d
trying to download
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/net/http.rb:921:in `connect': Connection reset by peer - SSL_connect (Faraday::ConnectionFailed)
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/net/http.rb:921:in `block in connect'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/timeout.rb:66:in `timeout'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/net/http.rb:921:in `connect'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/net/http.rb:862:in `do_start'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/net/http.rb:851:in `start'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/net/http.rb:1373:in `request'
	from /Library/Ruby/Gems/2.0.0/gems/faraday-0.9.2/lib/faraday/adapter/net_http.rb:82:in `perform_request'
	from /Library/Ruby/Gems/2.0.0/gems/faraday-0.9.2/lib/faraday/adapter/net_http.rb:40:in `block in call'
	from /Library/Ruby/Gems/2.0.0/gems/faraday-0.9.2/lib/faraday/adapter/net_http.rb:87:in `with_net_http_connection'
	from /Library/Ruby/Gems/2.0.0/gems/faraday-0.9.2/lib/faraday/adapter/net_http.rb:32:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/faraday-cookie_jar-0.0.6/lib/faraday/cookie_jar.rb:22:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/faraday_middleware-0.10.0/lib/faraday_middleware/response_middleware.rb:30:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/faraday_middleware-0.10.0/lib/faraday_middleware/response_middleware.rb:30:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/faraday_middleware-0.10.0/lib/faraday_middleware/response_middleware.rb:30:in `call'
	from /Library/Ruby/Gems/2.0.0/gems/faraday-0.9.2/lib/faraday/rack_builder.rb:139:in `build_response'
	from /Library/Ruby/Gems/2.0.0/gems/faraday-0.9.2/lib/faraday/connection.rb:377:in `run_request'
	from /Library/Ruby/Gems/2.0.0/gems/faraday-0.9.2/lib/faraday/connection.rb:177:in `post'
	from /Library/Ruby/Gems/2.0.0/gems/spaceship-0.36.0/lib/spaceship/client.rb:445:in `block in send_request'
	from /Library/Ruby/Gems/2.0.0/gems/spaceship-0.36.0/lib/spaceship/client.rb:347:in `with_retry'
	from /Library/Ruby/Gems/2.0.0/gems/spaceship-0.36.0/lib/spaceship/client.rb:444:in `send_request'
	from /Library/Ruby/Gems/2.0.0/gems/spaceship-0.36.0/lib/spaceship/client.rb:385:in `request'
	from /Library/Ruby/Gems/2.0.0/gems/spaceship-0.36.0/lib/spaceship/portal/portal_client.rb:413:in `fetch_csrf_token_for_provisioning'
	from /Library/Ruby/Gems/2.0.0/gems/spaceship-0.36.0/lib/spaceship/portal/portal_client.rb:361:in `block in download_provisioning_profile'
	from /Library/Ruby/Gems/2.0.0/gems/spaceship-0.36.0/lib/spaceship/portal/portal_client.rb:449:in `ensure_csrf'
	from /Library/Ruby/Gems/2.0.0/gems/spaceship-0.36.0/lib/spaceship/portal/portal_client.rb:360:in `download_provisioning_profile'
	from /Library/Ruby/Gems/2.0.0/gems/spaceship-0.36.0/lib/spaceship/portal/provisioning_profile.rb:333:in `download'
```

Hope this helps.

**Comment by arunks** [comment]
Thanks @KrauseFx updating ruby fixes this.

**Comment by julestburt** [comment]
thx - was solved with:
brew install ruby
gem install fastlane

**Comment by sungjin0213** [comment]
If you run into issue of "brew" hanging, then you might want to try:
/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"

It seems like core package somehow got messed up in my case, and all brew commands were hanging.

**Comment by KrauseFx** [comment]
Thanks everyone for your help resolving this. @mfurtak pushed a new release of _fastlane_core_, which will print out the following: https://github.com/fastlane/fastlane/pull/6593

There is no other way than to update or re-install Ruby, as OpenSSL is compiled into your Ruby environment. Let us know if you have any more issues 👍

**Comment by jeremangnr** [comment]
Just a comment, after running `brew install ruby` I had to restart my terminal instance so changes got propagated correctly (`ruby -v` was still saying 2.0.0 before restarting it). Then everything worked OK for me

**Comment by jacobvanorder** [comment]
🐄  Holy cow. 🐄 

For anyone using Bamboo as their CI solution, I had to install [Bamboozled](https://marketplace.atlassian.com/plugins/com.alienfast.bamboozled.ruby.bamboozled-ruby-plugin/server/overview), a Bamboo plugin, to get Bamboo to use something **other** than the system Ruby. 

This plugin will use Bundler along with Bundler `exec`. It was a combination of this that allowed me to use Ruby 2.3.1 along with fastlane.

**Comment by chappejw** [comment]
I had luck getting all my fastlane tools to work again after mysteriously encountering this error after making use of a web debugger named Charles ( which seemed to have changed my OpenSSL )

The step that made the difference for me was in symlinking my Open SSL so I was using the correct version

    `ln -s /usr/local/Cellar/openssl/1.0.2j/bin/openssl /usr/local/bin/openssl`

found at this site
http://tonyle.ca/blog/?p=101

**Comment by TKBurner** [comment]
@chappejw @jacobvanorder Glad to hear you guys got up and running. Let me know if there are any changes we can make to improve this.

**Comment by scottrhoyt** [comment]
For those of you attempting the `brew install ruby` method and when you `gem install fastlane` you get the error:

```
ERROR: While executing gem ... (TypeError)
no implicit conversion of nil into String
```

Try updating your rubygems system with `gem update --system`.

**Comment by ivanicspeter92** [comment]
+1

**Comment by jlubeck** [comment]
I know this is not fastlane specific but I'm running out of options. I managed to update everything These are my current versions:

```
jan$ ruby -v
ruby 2.3.1p112 (2016-04-26 revision 54768) [x86_64-darwin16
jan$ openssl version
OpenSSL 1.0.2j  26 Sep 2016
```

And I'm still getting the same `Connection reset by peer - SSL_connect error`.

Any idea what else I might try?

Thank you

**Comment by geraldWilliam** [comment]
@jlubeck If you're using rbenv, I made sure that my openssl and ruby were on the correct versions and still got the "Connection reset by peer error" until I ran `rbenv rehash`. That fixed it.

**Comment by jlubeck** [comment]
@geraldWilliam still no luck for me :(

**Comment by jlubeck** [comment]
mmm that's weird.... didn't do anything new after my last post, and now it's working!

**Comment by ivanicspeter92** [comment]
My issues were resolved by executing the following commands
`brew update && brew install ruby
sudo gem install fastlane
sudo gem update produce
sudo gem update sigh`

**Comment by TKBurner** [comment]
@ivanicspeter92 Awesome! Glad to hear you got up and running.

**Comment by danieljimenez** [comment]
Is it not possible to get this working with the macOS system ruby? I'm running El Cap and am running into this issue.

**Comment by chappejw** [comment]
A couple key steps which are not mentioned here are that after updating your Ruby to version 2.3.1 etc ( at the time of this writing ), you will need to uninstall all of your ruby gems and re-install them to ensure everything is compatible, make sure you set your path to the new ruby in your .bash_profile or .bashrc, then re-install executable-hooks with the following command 

`sudo gem install --user-install executable-hooks`

**Comment by TKBurner** [comment]
@danieljimenez On El Cap you should be able to get it to work with system ruby following the steps by @ivanicspeter92. You might need to use bundler (like `bundle exec fastlane beta`) in some cases. Let me know how it goes.

@chappejw Great point, are you using a tool to manage ruby like rvm or rebnev?

**Comment by danieljimenez** [comment]
@TKBurner running `brew install ruby` is no longer the system ruby. I mean the shipping ruby with macOS.

**Comment by TKBurner** [comment]
@danieljimenez Sorry, I meant the last three commands, and `xcode-seclect --install`.  Do you still get errors? Can you run `fastlane env` and past the output? Thanks so much 👍

**Comment by marbetschar** [comment]
@TKBurner I'm running into the exact same issue running El Cap on the CI machine. Running locally using macOS Sierra everything works as expected.

**Comment by marbetschar** [comment]
@geraldWilliam @TKBurner using `rbenv rehash` did the trick! Thank you!!!!

**Comment by TKBurner** [comment]
@marbetschar Awesome! So glad to hear you got up and running. Always feel free to reach out. 👍

**Comment by kashifangus** [comment]
I was running El Capitan 10.11.6 and ran into this problem today too.

I just installed brew and did:
- brew update
- brew install ruby
Ran my CI build script and it worked.

I should say that my fastlane tools (and other gems) are part of my gem file that get installed/updated via bundler every time I do a build, so I didn't have to manually update those gems.

**Comment by TKBurner** [comment]
@cogentParadigm We are still looking into a better long term solution. Expect and update on this thread in the coming week. Thanks!

**Comment by jostster** [comment]
I'm having this exact same issue. I have updated system ruby, installed rbenv and rvm and still nothing is working. My `fastlane env` is 

```
<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                 | Value                                         |
| ------------------- | --------------------------------------------- |
| OS                  | 10.11.6                                       |
| Ruby                | 2.3.1                                         |
| Bundler?            | false                                         |
| Xcode Path          | /Applications/Xcode.app/Contents/Developer/   |
| Xcode Version       | 8.1                                           |
| Git                 | git version 2.9.3 (Apple Git-75)              |
| Installation Source | /Users/bamboo-agent/.gem/bin/fastlane         |
| Host                | Mac OS X 10.11.6 (15G1108)                    |
| Ruby Lib Dir        | /Users/bamboo-agent/.rbenv/versions/2.3.1/lib |
| OpenSSL Version     | OpenSSL 1.0.2j  26 Sep 2016                   |


### fastlane files:

**No Fastfile found**


**No Appfile found**


### fastlane gems

| Gem                 | Version | Update-Status |
| ------------------- | ------- | ------------- |
| credentials_manager | 0.16.2  | ✅ Up-To-Date  |
| spaceship           | 0.37.0  | ✅ Up-To-Date  |
| fastlane_core       | 0.53.0  | ✅ Up-To-Date  |
| deliver             | 1.15.0  | ✅ Up-To-Date  |
| snapshot            | 1.16.3  | ✅ Up-To-Date  |
| frameit             | 3.0.0   | ✅ Up-To-Date  |
| pem                 | 1.4.0   | ✅ Up-To-Date  |
| cert                | 1.4.3   | ✅ Up-To-Date  |
| sigh                | 1.11.2  | ✅ Up-To-Date  |
| produce             | 1.3.0   | ✅ Up-To-Date  |
| gym                 | 1.12.0  | ✅ Up-To-Date  |
| pilot               | 1.12.0  | ✅ Up-To-Date  |
| scan                | 0.14.1  | ✅ Up-To-Date  |
| supply              | 0.7.1   | ✅ Up-To-Date  |
| match               | 0.10.0  | ✅ Up-To-Date  |
| screengrab          | 0.5.5   | ✅ Up-To-Date  |
| fastlane            | 1.107.0 | ✅ Up-To-Date  |


### Loaded fastlane plugins:

**No plugins Loaded**


<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.0.0        |
| highline                  | 1.7.8        |
| commander                 | 4.4.0        |
| terminal-table            | 1.4.5        |
| json                      | 1.8.3        |
| multipart-post            | 2.0.0        |
| faraday                   | 0.9.2        |
| faraday_middleware        | 0.10.0       |
| dotenv                    | 2.1.1        |
| net-ssh                   | 3.2.0        |
| net-sftp                  | 2.1.2        |
| plist                     | 3.1.0        |
| rubyzip                   | 1.1.7        |
| security                  | 0.1.3        |
| krausefx-shenzhen         | 0.14.10      |
| slack-notifier            | 1.5.1        |
| i18n                      | 0.7.0        |
| thread_safe               | 0.3.5        |
| tzinfo                    | 1.2.2        |
| minitest                  | 5.8.3        |
| activesupport             | 4.2.6        |
| CFPropertyList            | 2.3.3        |
| claide                    | 1.0.1        |
| colored                   | 1.2          |
| nanaimo                   | 0.2.2        |
| xcodeproj                 | 1.4.1        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.4        |
| terminal-notifier         | 1.7.1        |
| addressable               | 2.4.0        |
| credentials_manager       | 0.16.2       |
| multi_xml                 | 0.5.5        |
| unf_ext                   | 0.0.7.2      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20161021 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 1.6.8        |
| xcode-install             | 2.0.8        |
| word_wrap                 | 1.0.0        |
| multi_json                | 1.12.1       |
| babosa                    | 1.0.2        |
| excon                     | 0.54.0       |
| gh_inspector              | 1.0.2        |
| fastlane_core             | 0.53.0       |
| bundler                   | 1.13.6       |
| mini_magick               | 4.5.1        |
| xcpretty-travis-formatter | 0.0.4        |
| uber                      | 0.0.15       |
| representable             | 2.3.0        |
| retriable                 | 2.1.0        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| hurley                    | 0.2          |
| little-plugger            | 1.1.4        |
| logging                   | 2.1.0        |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.2.4      |
| google-api-client         | 0.9.19       |
| io-console                | 0.4.5        |
</details>


*generated on:* **2016-11-08**
</details>
```

which clearly says ruby is now 2.3.1, but the SSL_Connect error still says RUBY_VERSION is 2.0.0. I have even rebooted the machine.

**Comment by ohwutup** [comment]
Hey everyone, we've been working on a new way to install and use fastlane, all you have to do is to download `fastlane.zip` and double click the `install` file. This will install fastlane with all its Ruby and OpenSSL dependencies, and should all work out of the box.
We just finished building this, and would love your early feedback on this beta. Please let us know how it works for you and if you run into any problems :+1: 
Download the latest version [here](http://kits-crashlytics-com.s3.amazonaws.com/fastlane/standalone/bundle-1.107.0.zip).

**Comment by wema1043** [comment]
@ohwutup I tried your solution, but still getting same isue

**Comment by ohwutup** [comment]
Hey @wema1043 could you share your output? And also confirm that `which fastlane` points to `~/.fastlane/...` instead of a gem-installed version?

**Comment by wema1043** [comment]
Hei @ohwutup,
I Tried your solution with newest fastlane version from your `.zip` 1h later and everything worked fine.
Don't know what happened before.

**Comment by samingle** [comment]
Had all sorts of trouble getting this working following the homebrew-based instructions from the error output. If you were previously running the system ruby, and you upgraded to homebrew's ruby using the provided instructions, **and** you're using bundler to manage fastlane, you'll need to reinstall bundler once your ruby version has been upgraded (`sudo gem install bundler` did the job for me), otherwise fastlane won't see the new ruby version.

**Comment by leoniralves** [comment]
What worked for me was to update Ruby for latest vesion (2.3.3), openssl (1.0.2) and reinstall Fastlane.
But I needed to restart my system and manually set PATH of updated openssl.
http://tonyle.ca/blog/?p=101

**Comment by KrauseFx** [comment]
We updated our [installation instructions](https://docs.fastlane.tools/#choose-your-installation-method) to also include [fastlane.zip](https://download.fastlane.tools), which solves this problem 🚀

**Comment by RishabhTayal** [comment]
I tried installing the latest version of `fastlane` through zip. I tried @leoniralves solution (https://github.com/fastlane/fastlane/issues/6553#issuecomment-265428621) . I tried @KrauseFx solution (https://github.com/fastlane/fastlane/issues/6553#issuecomment-254263591). Still getting this issue.

**Comment by RishabhTayal** [comment]
Solved the issue. I tried installing ruby several times. The issue was related to something on my git server. Apparently there was some ssl issue with the git url.

**Comment by KrauseFx** [comment]
I'll lock the conversation for this issue now, as there is a very high number of people subscribed. Please submit a new issue if you still experience this problem.
